backend/src/main/java/com/rota/api/StaffController.java

```diff
@@ -0,0 +1,75 @@
+package com.rota.api;
+
+import com.rota.api.dto.EngagementDto;
+import com.rota.auth.AuthenticationUtils;
+import com.rota.database.orm.engagement.EngagementRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Api(value = "Manages the responses to staff-related queries")
+public class StaffController {
+  @Autowired
+  EngagementRepository engagementRepository;
+
+  private List<EngagementDto> getStaffEngagementsBetween(int staffId, Instant start, Instant end) {
```

Comment:
Once we get more methods like this which are used by the get and post endpoints we could probably move them into a `StaffManager` class, or something similar, so as to only have endpoints in this file, keeping it tidy

Reply:
In my previous project we used a `StaffService` class to handle any business logic which was then much easier to test.